tools/gen_usb_descriptor.py

```diff
@@ -21,6 +21,7 @@
                     help='length needed for the serial number in digits')
 parser.add_argument('--output_c_file', type=argparse.FileType('w'), required=True)
 parser.add_argument('--output_h_file', type=argparse.FileType('w'), required=True)
+parser.add_argument('--reduced_endpoint_mode', nargs='?',const=0, type=int)
```

Comment:
I'd suggest passing in the board, mcu and port. reduced_endpoint_mode is a not specific enough to make the logic within here understandable.

Reply:
That would make the conditionals board specific. That might not be a bad thing, as the low end F4 boards might be the only ones that every encounter this issue, but it's not super modular if that changes. Would it be better to pass in what endpoints to disable in particular, and have that specified in mpboardconfig?

Reply:
For now, I've changed the flag here and in the mpboardconfig to `USB_CDC_AND_MSC_ONLY` so it's more descriptive of what this actually does. I think it's worth exploring whether we want to go more modular than that in the future.